.github/scripts/ai-diagnose-failure.py

```
@@ -0,0 +1,129 @@
+SUMMARY_PATH = os.environ.get("GITHUB_STEP_SUMMARY", "")
+WORKFLOW_NAME = os.environ.get("WORKFLOW_NAME", "E2E job")
+RUN_URL = os.environ.get("RUN_URL", "")
+
+LOG_PATTERN = re.compile(r"error|traceback|panic|failed|exception", re.IGNORECASE)
+
+
+def extract_junit_failures(path):
+    if not path or not os.path.isfile(path):
+        return "(no junit.xml found)"
+    try:
+        tree = ET.parse(path)
```

Ruff (0.16.3)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)